### DISCUSSION TITLE
A few questions about "actuator"

### DISCUSSION THREAD

**Discussion by 5nevergiveup** [mdolab/dafoam#721] [General]
Hi guys, I have a few questions about "actuator" and I can see that in test code there is this: 

> ```
> "designVar": {
> "shapey": {"designVarType": "FFD"},
> "uin": {"designVarType": "BC", "patches": ["inlet"], "variable": "U", "comp": 0},
> "actuator": {"actuatorName": "disk1", "designVarType": "ACTD"},
> },
> }
> ```

And 

```
> "designVar": {
> "shapey": {"designVarType": "FFD"},
> "alpha": {"designVarType": "AOA", "patches": ["inout"], "flowAxis": "x", "normalAxis": "y"},
> "actuator": {"actuatorName": "disk1", "designVarType": "ACTD", "comps": [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]},
> },
```

What does this "comps" stand for?  What does it affect? Because I don't see much difference in other things, including the definition of  `actuator(val, geo)`
I don't know what corresponds to `scale`,`adjustThrust` and `targetThrust` in actuatorDisk, actuatorPoint and actuatorLine respectively. Can you tell me more about what they mean? Also, when starting a project, how do you determine the `scale`, `adjustThrust`, and `targetThrust`? Thank you so much!

**Comment by friedenhe** [comment]
By default, the ACTD will have 10 design variables, including the location, direction, magnitude of the actuator disk. If you don't want to use that many design variables, you can prescribe a component index: comps. comps: [0] means the design variable is the x location of the actuator center. adjustThrust means you ask DAFoam to automatically adjust the scaling to match the targetThrust (in N) you prescribe.

**Comment by 5nevergiveup** [comment]
Thank you very much for your reply. I still have a few questions. 1. What integral does scale correspond to? I kind of didn't get it. 2. Can hybrid mode be used in DAPimpleFoam? 3. Can DAPisoFoam use reverse mode instead of fd mode?

**Comment by friedenhe** [comment]
1. I am not sure what "scale" you referred to. 2. No. the hybrid mode is outdated and is no longer supported... 3. Please use DAPimpleFoam (it uses the reverse mode and can run in the piso. mode) instead of DAPisoFoam..
